### ISSUE TITLE
Use large dataset

### ISSUE THREAD

**Issue by Maxfashko** [fchollet/keras#6980]
Hello. I had a problem with using a large dataset for semantic segmentation (more than 500k images 256 * 256). How do I load large amounts of data?
Is it possible to use this approach?

def get_image(count):
	x_train = np.load('imgs_train{part}.npy'.format(part=count))
	y_train = np.load('mask_train{part}.npy'.format(part=count))
        return x_train, y_train

for e in range(10):
	print("epoch %d" % e)
	x_train, x_train = get_image(1) # these are chunks of ~10k pictures
	model.fit(x_train, x_train, batch_size=32, nb_epoch=1)
	x_train, x_train = get_image(2) # these are chunks of ~10k pictures
	model.fit(x_train, x_train, batch_size=32, nb_epoch=1, validation_data=(x_test, y_test))

**Comment by prakritidev** [comment]
try using chunks of data rather than full dataset and use partial_fit method. 

This might help with you.

**Comment by Dref360** [comment]
I guess we need a guide for fit_generator. 
A lot of people are asking this kind of questions lately.

**Comment by ManishSahu53** [comment]
Use this kind of code

image_generator = train_datagen.flow_from_directory(
        'F:/Aerial Buiding Dataset/raw/train')

mask_generator = mask_datagen.flow_from_directory(
    'F:/Aerial Buiding Dataset/raw/mask')

**Comment by Maxfashko** [comment]
The problem is that I can not use the standard generator, since the size of the masks differs from the size of the original fragments (256 * 256 vs. 224 * 224). This is due to the fact that I use the Crop2D layer before network prediction to remove poorly predicted boundaries. So I implemented the following code. I hope this will work as I expect it to be:

`for e in range(int(epoch_ / (count_mini_batches))):
		for i in range(count_mini_batches):
			X_train, y_train = load_mini_batch(i)

			print("-"*30)
			print("local_epochs = "+str(i)+":"+str(count_mini_batches)+"    GLOBAL EPOCH = " + str(e) + ":" + str(int(epoch_ / (count_mini_batches))))
			print("-"*30)

			# Validate every 6 local epochs
			if i % 6 == 0:
				history = model.fit(X_train, y_train,
									batch_size=batch_sizes,
									epochs=1,
									verbose=1,
									shuffle=True,
									validation_data=(x_val, y_val))
                                # Save the weight of the model if the validation accuracy is the best
				clb_save_weights_the_best(history, suffix, model)
				save_history(history, suffix)

			else:
				history = model.fit(X_train, y_train,
									batch_size=batch_sizes,
									epochs=1,
									verbose=1,
									shuffle=True)

			del X_train
			del y_train`
